dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImplTest.java

```diff
@@ -363,4 +393,92 @@ public void testDeleteVersion_KeepPermissions() throws DotSecurityException, Dot
         }
     }
 
+    @Test
+    @UseDataProvider("dateTimeTestCases")
+    public void testIndexSearchFilteringByDateTime(final TestCase testCase){
+
+        LocalDateTime today, yesterday, tomorrow;
+
+        final LocalDateTime now = LocalDateTime.now();
+
+        today = now;
+        yesterday = today.minusDays(1).withHour(0).withMinute(0).withSecond(0);
+        tomorrow = today.plusDays(1).withHour(23).withMinute(59).withSecond(59);
+
+        buildLuceneQueryAndTest(testCase, today, yesterday, tomorrow);
+    }
+
+    @Test
+    @UseDataProvider("timeTestCases")
+    public void testIndexSearchFilteringByTime(final TestCase testCase){
+
+        LocalDateTime today, yesterday, tomorrow;
+
+        final LocalDateTime now = LocalDateTime.now();
+        today = LocalDateTime.of(1970, 1, 1, now.getHour(), now.getMinute(), now.getSecond());
+        yesterday = LocalDateTime.of(1970, 1, 1, 0, 0, 0);
+        tomorrow = LocalDateTime.of(1970, 1, 1, 23, 59, 59);
+
+        buildLuceneQueryAndTest(testCase, today, yesterday, tomorrow);
+    }
+
+    /**
+     *
+     * @param testCase
+     * @param today
+     * @param yesterday
+     * @param tomorrow
+     */
+    private void buildLuceneQueryAndTest(final TestCase testCase, final LocalDateTime today,
+            final LocalDateTime yesterday, final LocalDateTime tomorrow) {
+        final ContentType contentType = TestDataUtils.getNewsLikeContentType();
+        Contentlet contentlet = createNewsContent(contentType, today);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'contentlet' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4545506)